google/oauth2/service_account.py

```diff
@@ -418,6 +389,18 @@ def _make_authorization_grant_assertion(self):
 
     @_helpers.copy_docstring(credentials.Credentials)
     def refresh(self, request):
+        if (
+            self._universe_domain != _DEFAULT_UNIVERSE_DOMAIN
+            and not self._jwt_credentials
+        ):
+            raise exceptions.RefreshError(
+                "self._jwt_credentials is missing for non-default universe domain"
+            )
+        if self._universe_domain != _DEFAULT_UNIVERSE_DOMAIN and self._subject:
```

Comment:
nit: I think collapsing `self._universe_domain != _DEFAULT_UNIVERSE_DOMAIN ` into a single if statement, and then nesting the following conditions might improve readability.

Reply:
I initially wrote the way as you said, but it seems the coverage util has some issue, it complains some part of the code missed coverage even though I have the tests. Changing to the current way solved the issue, lol

Reply:
What a pain lol